### PR TITLE
feat: thorchain LP withdraw outbound fees

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Adds withdraw outbound fees as protocol fees for pool assets, estimating the Tx cost of sending from THOR address (contract/inbound address) to the user's address.

Note: As explained in the comments, this does not take RUNE into account in this PR and only relates to the pool asset for now - RUNE has constant 0.02 RUNE fees and has not been implemented here for the sake of simplicity and avoiding branching. 

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- relates to https://github.com/shapeshift/web/issues/6413, by having protocol fees (outbound fees), we can simply *4 said outbound fee which will give us the minimum withdraw amount. See https://gitlab.com/thorchain/thornode/-/blob/develop/x/thorchain/querier_quotes.go#L490

Low - ensure withdraws still behave as expected, there are no errors with estimations, and the user-currency fee looks sane

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- When withdrawing RUNE asymetrically, there is no protocol fee row displayed
- When withdrawing either asset asym, or symetrically, the protocol fee looks sane as outbound fee (i.e a native/token send on the destination chain)

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Native EVM assets

Screenshot 2024-03-12 at 14 23 02
Screenshot 2024-03-12 at 14 22 45

- Tokens

Screenshot 2024-03-12 at 14 23 44

- UTXOs

Screenshot 2024-03-12 at 14 23 14

- ATOM

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
* **#6415** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6415?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/Sweep.tsx [hunk 1]
```diff
@@ -194,7 +194,7 @@ export const Sweep = ({
             <Row.Value>
               <Skeleton isLoaded={isEstimatedFeesDataSuccess}>
                 <Box textAlign='right'>
-                  <Amount.Fiat value={estimatedFeesData?.txFeeFiat ?? '0'} />
+                  <Amount.Fiat value={estimatedFeesData?.txFeeFiatUserCurrency ?? '0'} />
```
Comment (gomesalexandre) [inline src/components/Sweep.tsx:197]: LSP renamed to honor the `userCurrency` terminology

File: src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx [hunk 2]
```diff
@@ -564,7 +564,7 @@ export const BorrowInput = ({
                   <Skeleton
                     isLoaded={Boolean(isEstimatedSweepFeesDataSuccess && estimatedSweepFeesData)}
                   >
-                    <Amount.Fiat value={estimatedSweepFeesData?.txFeeFiat ?? '0'} />
+                    <Amount.Fiat value={estimatedSweepFeesData?.txFeeFiatUserCurrency ?? '0'} />
```
Comment (gomesalexandre) [inline src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx:567]: Ditto LSP rename

File: src/pages/Lending/hooks/useGetEstimatedFeesQuery.ts [hunk 3]
```diff
@@ -5,21 +5,38 @@ import type { EstimateFeesInput } from 'components/Modals/Send/utils'
...
   // These should not be undefined when used with react-query, but may be when used outside of it since there's no "enabled" option
-  if (!asset || !estimateFeesInput?.to || !estimateFeesInput.accountId) return
+  if (
+    !asset ||
+    !estimateFeesInput?.to ||
+    !(estimateFeesInput.accountId || estimateFeesInput.pubkey)
+  )
+    return
+
+  const feeAsset = selectFeeAssetById(store.getState(), asset.assetId)
```
Comment (gomesalexandre) [inline src/pages/Lending/hooks/useGetEstimatedFeesQuery.ts:27]: This was 100% wrong previously - fees are in native asset and we would get them wrong by orders of magnitude for tokens

File: src/pages/ThorChainLP/components/RemoveLiquidity/RemoveLiquidityInput.tsx [hunk 4]
```diff
@@ -363,6 +364,51 @@ export const RemoveLiquidityInput: React.FC<RemoveLiquidityInputProps> = ({
...
+    if (!inboundAddressesData) return undefined
+
+    const amountCryptoPrecision = actualAssetWithdrawAmountCryptoPrecision
+
+    return {
+      amountCryptoPrecision,
+      assetId: poolAsset.assetId,
+      to: poolAssetAccountAddress,
+      sendMax: false,
+      // Router holds tokens, inbound address holds native asset
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/RemoveLiquidity/RemoveLiquidityInput.tsx:383]: Another day another weirdness at protocol-level, gotta do what you gotta do to make things work

**Comment by gomesalexandre** [comment]
Moving to draft for @kaladinlight to cleanup and/or close

**Comment by kaladinlight** [comment]
https://github.com/shapeshift/web/pull/6423 this covers protocol fees. it looks like there is some extra cleanup in here that can be broken out if desired, but outbound fees are accounted for as it stands

**Pull request closed without merging**